core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -154,12 +181,93 @@
       // Create masters.
       mRegistry = new MasterRegistry();
       mSafeModeManager = new DefaultSafeModeManager();
-      MasterUtils.createMasters(mJournalSystem, mRegistry, mSafeModeManager);
+      mStateLock = new ReentrantReadWriteLock();
+      MasterUtils.createMasters(mRegistry,
+          new MasterContext(mJournalSystem, mSafeModeManager, mStateLock.readLock()));
     } catch (Exception e) {
       throw new RuntimeException(e);
     }
   }
 
+  @Override
+  public String exportJournal(ExportJournalOptions options) throws IOException {
+    AlluxioURI uri = new AlluxioURI(options.getTargetDirectory());
+    LOG.info("Exporting journal backup to {}", uri);
+    UnderFileSystem ufs = UnderFileSystem.Factory.create(uri.toString());
```

Comment:
I think this will cause problems. I don't think we should ever call `create(uri)` directly. We should always create ufs with a configuration. This create must be associated with configs somehow.

Reply:
Updated to either use `createForRoot()` or `create("/", UnderFileSystemConfiguration.defaults())`. Also updated the API so that users can use a boolean argument to determine whether to export to the root UFS vs local filesystem.